inst/stan/functions/delay_lpmf.stan

```diff
@@ -27,33 +27,59 @@ real delay_group_lpmf(array[] int groups, int start, int end, array[] int obs,
                       array[,] int rdlurd, vector srdlh, matrix ref_lh,
                       array[] int dpmfs, int ref_p, int rep_h, int ref_as_p,
                       array[] real phi, int model_obs, int model_miss,
-                      vector miss_ref_lprop) {
-  // For missing model
-  // Passing in missing prop
-  // Pass in missing obs
-  // Make model optional
-  // need to know which report date data is linked (i.e we need a new vector
-  // look-up) by group
-  // sum by this look-up to get cases by report date and group (can also make
-  // this a matrix)
-  // scale all data by missing proportion or 1 minus missing proportion
-
+                      int miss_obs, array[] int missing_reference,
+                      array[,] int obs_by_report, vector miss_ref_lprop,
+                      array[] int sdmax, array[] int csdmax) {
   // Where am I?
   real tar = 0;
   int i_start = ts[1, start];
   int i_end = ts[t, end];
   array[3] int n = filt_obs_indexes(i_start, i_end, csl, sl);
-  array[n[3]] int filt_obs = obs[(n[1] + 1):n[2]];
+  array[n[3]] int filt_obs = segment(obs, n[1] + 1, n[3]);
+  // What is going to be used for storage
   vector[n[3]] log_exp_obs;
+  vector[model_miss ? miss_obs : 0]  log_exp_miss_ref;
 
   // Combine expected final obs and date effects to get expected obs
-  log_exp_obs = expected_obs_from_snaps(
-    i_start, i_end, imp_obs, rdlurd, srdlh, ref_lh, dpmfs, ref_p, rep_h, ref_as_p, sl, csl, sg, st, n[3]
-  );
+  // If missing reference module in place calculated all expected obs vs 
+  // just those observed and allocate if missing or not.
+  if (model_miss) {
+    array[3] int f = filt_obs_indexes(i_start, i_end, csdmax, sdmax);
+    vector[f[3]] log_exp_complete;
 
+    // Calculate all expected observations
+    log_exp_complete = expected_obs_from_snaps(
+      i_start, i_end, imp_obs, rdlurd, srdlh, ref_lh, dpmfs, ref_p, rep_h, ref_as_p, sdmax, csdmax, sg, st, f[3]
+    );
+
+    // Allocate to just those actually observed
+    log_exp_obs = allocate_observed_obs(
+      i_start, i_end, log_exp_complete, sl, csl, sdmax, csdmax
+    );
+    log_exp_obs = apply_missing_reference_effects(
+      i_start, i_end, log_exp_obs, sl, csl, log1m(miss_ref_lprop)
```

Comment:
Should this not be `log1m_exp`? as `miss_ref_lprop` is already on the log scale?

Reply:
yes it should good spot - it was what causing the initial not working at all issues. I fixed it locally a few commits not at all clear why its showing up not changed here...

Reply:
if I just failed to save the function file that is a lot of time wasted and quite embarrassing!

Reply:
Would you necessarily expect perfect parameter recovery? There are quite a few small numbers in the toy data set perhaps leading to edge cases given we're using a continuous approximation with a fixed multiplier for missing data. In other words, the simulated data is not from the data generating process of the model - though you could try that (fixed scaling rather than binomial sampling to generate the simulated data set).

Reply:
moving to main comment section for better threading ettiquette.